mod/beacon/validator/service.go

```diff
@@ -66,41 +72,55 @@ type Service[
 	// blobFactory is used to create blob sidecars for blocks.
 	blobFactory BlobFactory[
 		BeaconBlockT, BeaconBlockBodyT, BlobSidecarsT,
+		DepositT, Eth1DataT, ExecutionPayloadT,
 	]
 	// bsb is the beacon state backend.
-	bsb StorageBackend[BeaconStateT, *types.Deposit, DepositStoreT]
+	bsb StorageBackend[
+		BeaconBlockHeaderT, BeaconStateT, DepositT,
+		DepositStoreT, ExecutionPayloadHeaderT,
+	]
 	// blobProcessor is used to process blobs.
 	blobProcessor BlobProcessor[BlobSidecarsT]
 	// stateProcessor is responsible for processing the state.
 	stateProcessor StateProcessor[
 		BeaconBlockT,
+		BeaconBlockHeaderT,
 		BeaconStateT,
 		*transition.Context,
+		ExecutionPayloadHeaderT,
 	]
 	// localPayloadBuilder represents the local block builder, this builder
 	// is connected to this nodes execution client via the EngineAPI.
 	// Building blocks are done by submitting forkchoice updates through.
 	// The local Builder.
-	localPayloadBuilder PayloadBuilder[BeaconStateT, *types.ExecutionPayload]
+	localPayloadBuilder PayloadBuilder[BeaconStateT, ExecutionPayloadT]
 	// remotePayloadBuilders represents a list of remote block builders, these
 	// builders are connected to other execution clients via the EngineAPI.
-	remotePayloadBuilders []PayloadBuilder[BeaconStateT, *types.ExecutionPayload]
```

Comment:
note for myself. the validator service should just get a payload from the payload service, should the validator care about which builder it comes from? or should the payload service handle it?